src/main/java/com/parsing/pdf/parsing/ParsingPDFService.java

```diff
@@ -0,0 +1,238 @@
+    }
+
+    private String findStrippedTextFromFile(MultipartFile file) {
+        try {
+            PDDocument document = PDDocument.load(file.getBytes());
+            PDFTextStripper stripper = new PDFTextStripper();
+            String strippedText = stripper.getText(document);
+
+            if (strippedText.trim().isEmpty()) {
+                return extractTextFromScannedDocument(document);
+            }
+        } catch (Exception e) {
+            throw new PDFParsingException(String.format("Parsing for file: {} failed.", file.getOriginalFilename()));
+        }
+        return null;
+    }
+
+    public String extractTextFromScannedDocument(PDDocument document)
+            throws IOException, TesseractException {
+
+        // Extract images from file
+        PDFRenderer pdfRenderer = new PDFRenderer(document);
+        StringBuilder out = new StringBuilder();
+
+//        // Define the folder path
+//        Path userFilesFolderPath = Paths.get("tessdata");
+//
+//        // Create the folder if it doesn't exist
+//        if (!Files.exists(userFilesFolderPath)) {
+//            Files.createDirectory(userFilesFolderPath);
+//        }
+
+        ITesseract _tesseract = new Tesseract();
+        _tesseract.setVariable("textord_tabfind_find_tables", "1");
+//        _tesseract.setPageSegMode(7);
+//        _tesseract.setDatapath("C:/Users/Maksym.Fedosov/Documents/tessdata/");
+        _tesseract.setDatapath("E:/programming/projects/parsing/tessdata/");
+//        _tesseract.setDatapath("../..//teseract/tessdata/");
+//        _tesseract.setDatapath(userFilesFolderPath.toUri().toString());
+        _tesseract.setLanguage("ukr+eng");
+        log.info(document.getNumberOfPages() + "  pages in document");
+
+        for (int page = document.getNumberOfPages() - 1; page < document.getNumberOfPages(); page++) {
+            BufferedImage bim = pdfRenderer.renderImageWithDPI(page, 300, ImageType.GRAY);
+
+            // Create a temp image file
+            File temp = File.createTempFile("tempfile_" + page, ".png");
+            ImageIO.write(bim, "png", temp);
+
+            File png = new File("008.png");
+            ImageIO.write(bim, "png", png);
+
+            String fileTableName = "destination.png";
+            List<Rectangle> rectangles = rectangleDetector.detectRectangles(fileTableName);
+            final Mat table = Imgcodecs.imread(fileTableName);
+            for (Rectangle rectangle : rectangles) {
+                Mat mat = fillSquareWhitePixel(table, rectangle);
+                Imgcodecs.imwrite(rectangles.indexOf(rectangle) + ".png", mat);
+            }
+            for (Rectangle rectangle : rectangles) {
+                String filename = rectangles.indexOf(rectangle) + ".png";
+                log.info(filename);
+                String result = _tesseract.doOCR(new File(filename));
+                log.info(result);
+                out.append(result);
+            }
+            // Delete temp file
```

Comment:
this comment isn't necessary, from my opinion

Reply:
Done.